### PR TITLE
two chatgpt legos: create chat and estimate chat cost

### PR DESCRIPTION
**Pull request by dougsillars**
## Description
Please include a summary of the change, motivation and context.

lego 1:  takes a model, messages and temperature to create a chat.
lego2. takes the model & messages- and gives a cost estimate for the input query

### CODE REVIEW

**Review by shloka-bhalgat-unskript — APPROVED** [review]

File: ChatGPT/legos/chatgpt_create_chat/chatgpt_create_chat.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    )
+    temperature: float = Field(
+        ...,
+        description='The Temperature of the response.  Valid values are 0 to 2. Higher values will give more random results.',
+        title='temperature',
+    )
+
+
+def chatgpt_create_chat(handle, model: str, messages:list, temperature:float)-> Dict:
+    response = handle.ChatCompletion.create(
```
Comment (amit-chandak-unskript) [inline ChatGPT/legos/chatgpt_create_chat/chatgpt_create_chat.py:32]: You are not using temperature in the create call

File: ChatGPT/legos/chatgpt_create_chat/chatgpt_create_chat.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+
+class InputSchema(BaseModel):
+    messages: List = Field(
+        ..., description='The messages for Chat GPT', title='messages'
+    )
+    model: str = Field(
+        ..., description='The ChatGPT Model to use in your query.', title='model'
+    )
+    temperature: float = Field(
```
Comment (amit-chandak-unskript) [inline ChatGPT/legos/chatgpt_create_chat/chatgpt_create_chat.py:24]: Default value should be zero.

File: ChatGPT/legos/chatgpt_create_chat/chatgpt_create_chat.py (hunk 1 above)
Comment (amit-chandak-unskript) [inline ChatGPT/legos/chatgpt_create_chat/chatgpt_create_chat.py:32]: Shouldnt the messages be like
```
    messages = [{"role": "user", "content": prompt}]
```
where prompt is nothing but your messages input, as the user will not provide the prompt.

File: ChatGPT/legos/chatgpt_get_token_count/chatgpt_get_token_count.py [hunk 3]
```diff
@@ -0,0 +1,56 @@
+from __future__ import annotations
+
+##
+##  Copyright (c) 2023 unSkript, Inc
+##  All rights reserved.
+##
+#Note - this requires the tiktoken library to be installed
+from pydantic import BaseModel, Field
+import tiktoken
```
Comment (amit-chandak-unskript) [inline ChatGPT/legos/chatgpt_get_token_count/chatgpt_get_token_count.py:9]: Did we add this library to the requirements file in unskript?  Any new package needs to be added there.

File: ChatGPT/legos/chatgpt_get_token_count/chatgpt_get_token_count.py [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+              if key == "name":  # if there's a name, the role is omitted
+                  num_tokens += -1  # role is always required and always 1 token
+      num_tokens += 2  # every reply is primed with <im_start>assistant
+      return num_tokens
+  else:
+      raise NotImplementedError(f"""num_tokens_from_messages() is not presently implemented for model {model}.
+  See https://github.com/openai/openai-python/blob/main/chatml.md for information on how messages are converted to tokens.""")
+
+
+def unskript_default_printer(output):
```
Comment (amit-chandak-unskript) [inline ChatGPT/legos/chatgpt_get_token_count/chatgpt_get_token_count.py:47]: This should be renamed to the lego function name with printer suffix. like chatgpt_get_token_count_printer.

**Comment by amit-chandak-unskript** [comment]
We are not adding chatgpt connector, hence removing these legos

**Pull request closed without merging**